test/e2e/monitoring_stack_controller_test.go

```diff
@@ -60,35 +71,20 @@ func TestMonitoringStackController(t *testing.T) {
 
 func emptyStackCreatesPrometheus(t *testing.T) {
 	ms := newMonitoringStack("empty-stack")
+	defer f.K8sClient.Delete(context.Background(), ms)
```

Comment:
good find! 🙇 
How about `newFoobar(t, ..)`and each `newFoobar` calls `t.Cleanup(...)` ?

Reply:
Good question, I am not sure what's the best way to handle this. It's really a question of who should own the lifecycle management of an object: the `newX` function or the caller itself.

Reply:
For the tests at least, the default would be to cleanup resource it creates, wouldn't it? If we follow this pattern all resources created will be automatically deleted. What I haven't tested is if it would yield the same behaviour as it does now. I.E .. currently `empty-stack` will be deleted at the end of `emptyStackCreatesPrometheus` function whereas t.Cleanup may be deleting the resource at the end of the a test unless these methods are called within t.Run(...).

Reply:
btw, this doesn't block the patch from merging ... it was just a thought.

Reply:
Let's try out the `t.Cleanup` approach. It might help us keep tests independent by default.